tpu_inference/layers/vllm/fused_moe.py

```diff
@@ -124,25 +125,23 @@ def _gmm(lhs, rhs, rhs_scale, rhs_bias, group_sizes):
             group_offset=jnp.array(0),
         )
 
-    rhs_scale_spec = None if rhs_scale is None else P(None, None, None,
-                                                      "model")
-    rhs_bias_spec = None if rhs_bias is None else P(None, None, "model")
+    rhs_scale_spec = None if rhs_scale is None else P(
+        None, None, None, ShardingAxisName.MLP_TENSOR)
+    rhs_bias_spec = None if rhs_bias is None else P(
+        None, None, ShardingAxisName.MLP_TENSOR)
 
     gmm_result = shard_map(
         _gmm,
         mesh=mesh,
-        in_specs=(
-            P("data", None),
-            P(None, "model", None),
-            rhs_scale_spec,
-            rhs_bias_spec,
-            P("data"),
-        ),
-        out_specs=(P("data", "model")),
+        in_specs=(P(ShardingAxisName.MLP_DATA,
+                    None), P(None, ShardingAxisName.MLP_TENSOR,
+                             None), rhs_scale_spec, rhs_bias_spec,
+                  P(ShardingAxisName.MLP_DATA)),
+        out_specs=(P(ShardingAxisName.MLP_DATA, ShardingAxisName.MLP_TENSOR)),
         check_vma=False,
     )(lhs, rhs, rhs_scale, rhs_bias, group_sizes)
 
-    tp_size = mesh.shape["model"]
+    tp_size = mesh.shape['model'] * mesh.shape.get('attn_dp', 1)
```

Comment:
just to double check, I assume you couldn't use `ShardingAxisName` here because that could